cmd/snap-confine/mount-support.c

```diff
@@ -502,6 +476,29 @@ static bool __attribute__ ((used))
 	return false;
 }
 
+static int sc_open_snap_update_ns()
+{
+	char buf[PATH_MAX] = { 0 };
```

Comment:
Done. I'm worried now, should all path handling code use `PATH_MAX + 1` buffers?

Reply:
It depends on the api used. readlink() doesn't add a trailing '\0' so we need to account for it using all of buf, and then we come along and tack on '\0'. Because we are using sc_must_snprintf() down below, what you had wasn't a security issue-- sc_must_sprintf() would've die()d-- but this way in the off chance of a super long symlink that is exactly PATH_MAX, it won't die() unnecessarily.